File: src/Energy.Core/Entities/Commerce/CatalogItems/CatalogItem.cs
using Energy.Core.Extensions.CustomTypes;
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;

namespace Energy.Core.Entities.Commerce.CatalogItems
{
    public class CatalogItem : Entity
    {
        public string Name { get; set; }
        public string Description { get; set; }
        public string Brand { get; set; }
        public decimal? Price { get; set; }
        public decimal? VAT { get; set; }
        public ICollection<string> PictureUris { get; set; }
        public DateTime? CreatedAt { get; set; }
        public DateTime? AvailableFrom { get; set; }
        public Weight Weight { get; set; }
        public Dimension Dimension { get; set; }
        public IDictionary<string, string> Properties { get; set; }

        public CatalogCategory Category { get; set; }
        public virtual ICollection<Review> Reviews { get; set; } = new Collection<Review>();
    }
}

File: src/Energy.Web/ClientApp/src/store/catalog/CatalogItem.ts
import { Action, Reducer } from 'redux';
import { AppThunkAction } from '..';

export interface CatalogItemState {
    item: CatalogItem;
}

export interface CatalogItem {
    Id: string;
    Name: string;
    Description: string;
    Brand: string;
    Price: number;
    PictureUris: string[];
    CreatedAt: Date;
    AvailableFrom: Date;
    Weight: Weight;
    Dimension: Dimension;
    Properties: [];
    Category: Category;
    Reviews: Review[];
}

export interface Weight {

}

export interface Dimension {

}

export interface Category {

}

export interface Review {

}

export interface CreateCatalogItemAction { type: 'CREATE' }
export interface UpdateCatalogItemAction { type: 'UPDATE' }
export type KnownAction = CreateCatalogItemAction | UpdateCatalogItemAction;

export const actionCreators = {
    create: () => ({ type: 'CREATE' } as CreateCatalogItemAction ),
    update: () => ({ type: 'UPDATE' } as UpdateCatalogItemAction )
};
